### PR TITLE
[10.0] partner multi relation hierarchy

### PR DESCRIPTION
**Pull request by NL66278**
Adds possibility to add hierarchical relations.

### CODE REVIEW

**Comment by NL66278** [comment]
@Yajo Do you see any reason why this should keep the label 'Needs fixing'?

**Comment by yajo** [comment]
Nope, removed :blush:

File: partner_multi_relation_hierarchy/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8 -*-
```
Comment (nikul-serpentcs) [inline partner_multi_relation_hierarchy/__manifest__.py:1]: @NL66278 IMHO Please added a `website` and  URL as described in https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L10

File: partner_multi_relation_hierarchy/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    "complexity": "normal",
+    "category": "Customer Relationship Management",
+    "license": "AGPL-3",
+    "depends": [
+        "partner_multi_relation",
+    ],
+    "data": [
+        'views/res_partner.xml',
+        'views/res_partner_relation_type.xml',
+        'security/ir.model.access.csv',
```
Comment (nikul-serpentcs) [inline partner_multi_relation_hierarchy/__manifest__.py]: @NL66278 IMHO security file should be always first :smile:

File: partner_multi_relation_hierarchy/tests/test_partner_hierarchy.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Therp BV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openerp.exceptions import ValidationError
```
Comment (nikul-serpentcs) [inline partner_multi_relation_hierarchy/tests/test_partner_hierarchy.py]: Replace `openerp` to `odoo`

**Comment by NL66278** [comment]
@nikul-serpentcs I have made the requested changes. Thank you for your review!

File: partner_multi_relation_hierarchy/README.rst [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+The hierarchical 'chains' will be made visible on the partner.
+
+More info
+---------
+
+For further information, please visit:
+
+ * https://www.odoo.com/forum/help-1
+
+Known issues / Roadmap
```
Comment (nikul-serpentcs) [inline partner_multi_relation_hierarchy/README.rst]: @NL66278 Remove section if does not need.

File: partner_multi_relation_hierarchy/models/res_partner_relation_hierarchy.py [hunk 5]
```diff
@@ -0,0 +1,102 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Therp BV <https://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+"""Abstract model to show each relation from two sides."""
+from psycopg2.extensions import AsIs
+
+from openerp import api, fields, models
```
Comment (nikul-serpentcs) [inline partner_multi_relation_hierarchy/models/res_partner_relation_hierarchy.py]: Replace `odoo`

File: partner_multi_relation_hierarchy/models/res_partner_relation_type.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Therp BV <https://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openerp import fields, models
```
Comment (nikul-serpentcs) [inline partner_multi_relation_hierarchy/models/res_partner_relation_type.py]: replace `odoo`

**Comment by NL66278** [comment]
@nikul-serpentcs Also implemented your new remarks. Thanks again!

**Review by nikul-serpentcs — APPROVED** [review]
Code review LGTM

**Review by lfreeke — APPROVED** [review]
Functional test :+1:

**Pull request merged**